diff --git a/go.mod b/go.mod
index e955449..1fe04bc 100644
--- a/go.mod
+++ clash/go.mod
@@ -3,7 +3,6 @@ module github.com/Dreamacro/clash
 go 1.17

 require (
-	github.com/Dreamacro/go-shadowsocks2 v0.1.7
 	github.com/go-chi/chi/v5 v5.0.4
 	github.com/go-chi/cors v1.2.0
 	github.com/go-chi/render v1.0.1
@@ -30,3 +29,5 @@ require (
 	golang.org/x/text v0.3.6 // indirect
 	gopkg.in/yaml.v3 v3.0.0-20200313102051-9f266ea9e77c // indirect
 )
+
+replace github.com/Dreamacro/go-shadowsocks2 => ./go-shadowsocks2
